internal/services/desktopvirtualization/virtual_desktop_scaling_plan_resource.go

```diff
@@ -0,0 +1,524 @@
+								string(desktopvirtualization.SessionHostLoadBalancingAlgorithmDepthFirst),
+							}, false),
+						},
+
+						"ramp_up_minimum_hosts_percent": {
+							Type:         pluginsdk.TypeInt,
+							Optional:     true,
+							ValidateFunc: validation.IntBetween(0, 100),
+						},
+
+						"ramp_up_capacity_threshold_percent": {
+							Type:         pluginsdk.TypeInt,
+							Optional:     true,
+							ValidateFunc: validation.IntBetween(0, 100),
+						},
+
+						"peak_start_time": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringMatch(regexp.MustCompile(`^([0-1]?[0-9]|2[0-3]):[0-5][0-9]$`),
+								`The time must be in the format HH:MM.`),
+						},
+
+						"peak_load_balancing_algorithm": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(desktopvirtualization.SessionHostLoadBalancingAlgorithmBreadthFirst),
+								string(desktopvirtualization.SessionHostLoadBalancingAlgorithmDepthFirst),
+							}, false),
+						},
+
+						"ramp_down_start_time": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringMatch(regexp.MustCompile(`^([0-1]?[0-9]|2[0-3]):[0-5][0-9]$`),
+								`The time must be in the format HH:MM.`),
+						},
+
+						"ramp_down_load_balancing_algorithm": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(desktopvirtualization.SessionHostLoadBalancingAlgorithmBreadthFirst),
+								string(desktopvirtualization.SessionHostLoadBalancingAlgorithmDepthFirst),
+							}, false),
+						},
+
+						"ramp_down_minimum_hosts_percent": {
+							Type:         pluginsdk.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IntBetween(0, 100),
+						},
+
+						"ramp_down_capacity_threshold_percent": {
+							Type:         pluginsdk.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IntBetween(0, 100),
+						},
+
+						"ramp_down_force_logoff_users": {
+							Type:     pluginsdk.TypeBool,
+							Required: true,
+						},
+
+						"ramp_down_stop_hosts_when": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(desktopvirtualization.StopHostsWhenZeroActiveSessions),
+								string(desktopvirtualization.StopHostsWhenZeroSessions),
+							}, false),
+						},
+
+						"ramp_down_wait_time_minutes": {
+							Type:     pluginsdk.TypeInt,
+							Required: true,
+						},
+
+						"ramp_down_notification_message": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+						},
+
+						"off_peak_start_time": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringMatch(regexp.MustCompile(`^([0-1]?[0-9]|2[0-3]):[0-5][0-9]$`),
```

Comment:
Didn't even occur to me that was something you could do! I've done that now, thanks for the tip

Reply:
@tombuildsstuff could you check how I've implemented this ask? Now getting linting errors telling me that `schema.SchemaValidateFunc is deprecated: please use SchemaValidateDiagFunc`
I can't find a `StringMatch` function that returns a `SchemaValidateDiagFunc` rather than a `SchemaValidateFunc`

Am I missing something or do I need to put something around `StringMatch` to make is a `SchemaValidateDiagFunc`